hs/src/Reach/Eval.hs

```diff
@@ -80,9 +80,208 @@ compileDApp shared_lifts exports (SLV_Prim (SLPrim_App_Delay at top_s (top_env,
+  pan = mapM pan
+
+instance (Pandemic b) => Pandemic (Either a b) where
+  pan = mapM pan
+
+instance Pandemic DLStmts where
+  pan = mapM pan
+
+instance Pandemic DLExpr where
+  pan = \case
+    DLE_Arg at arg -> DLE_Arg at <$> pan arg
+    DLE_LArg at larg  -> DLE_LArg at <$> pan larg
+    DLE_Impossible at i err -> return $ DLE_Impossible at i err
+    DLE_VerifyMuldiv at cxt ct argl err -> do
+      r <- pan argl
+      return $ DLE_VerifyMuldiv at cxt ct r err
+    DLE_PrimOp at primop argl -> DLE_PrimOp at primop <$> pan argl
+    DLE_ArrayRef at arg1 arg2 -> DLE_ArrayRef at <$> pan arg1 <*> pan arg2
+    DLE_ArraySet at arg1 arg2 arg3 -> DLE_ArraySet at <$> pan arg1 <*> pan arg2 <*> pan arg3
+    DLE_ArrayConcat at arg1 arg2 -> DLE_ArrayConcat at <$> pan arg1 <*> pan arg2
+    DLE_ArrayZip at arg1 arg2 -> DLE_ArrayZip at <$> pan arg1 <*> pan arg2
+    DLE_TupleRef at arg1 i -> do
+      r <- pan arg1
+      return $ DLE_TupleRef at r i
+    DLE_ObjectRef at arg s -> do
+      r <- pan arg
+      return $ DLE_ObjectRef at r s
+    DLE_Interact at cxt slp s ty argl -> do
+      r <- pan argl
+      return $ DLE_Interact at cxt slp s ty r
+    DLE_Digest at argl -> DLE_Digest at <$> pan argl
+    DLE_Claim at cxt ct arg mbbs -> do
+      r <- pan arg
+      return $ DLE_Claim at cxt ct r mbbs
+    DLE_Transfer at arg1 arg2 marg -> DLE_Transfer at <$> pan arg1 <*> pan arg2 <*> pan marg
+    DLE_TokenInit at arg -> DLE_TokenInit at <$> pan arg
+    DLE_CheckPay at cxt arg marg -> DLE_CheckPay at cxt <$> pan arg <*> pan marg
+    DLE_Wait at targ -> DLE_Wait at <$> pan targ
+    DLE_PartSet at slp arg -> DLE_PartSet at slp <$> pan arg
+    DLE_MapRef at dlm arg -> DLE_MapRef at dlm <$> pan arg
+    DLE_MapSet at dlm arg marg -> DLE_MapSet at dlm <$> pan arg <*> pan marg
+    DLE_Remote at cxt arg ty s amt argl bill -> do
+      DLE_Remote at cxt arg ty s <$> pan amt <*> pan argl <*> pan bill
+    DLE_TokenNew at tknew -> DLE_TokenNew at <$> pan tknew
+    DLE_TokenBurn at arg1 arg2 -> DLE_TokenBurn at <$> pan arg1 <*> pan arg2
+    DLE_TokenDestroy at arg -> DLE_TokenDestroy at <$> pan arg
+    DLE_TimeOrder at margs_vars -> DLE_TimeOrder at <$> pan margs_vars
+    DLE_GetContract at -> return $ DLE_GetContract at
+    DLE_GetAddress at -> return $ DLE_GetAddress at
+    DLE_EmitLog at lk vars -> DLE_EmitLog at lk <$> pan vars
+    DLE_setApiDetails at who dom mc info -> return $ DLE_setApiDetails at who dom mc info
+    DLE_GetUntrackedFunds at marg arg -> DLE_GetUntrackedFunds at <$> pan marg <*> pan arg
+    DLE_FromSome at arg1 arg2 -> DLE_FromSome at <$> pan arg1 <*> pan arg2
+
+instance Pandemic DLVar where
+  pan (DLVar at m_locvar t i) = do
+    case m_locvar of
+      Nothing -> do
+        infections <- asks e_infections
+        r <- liftIO $ readIORef infections
+        case M.lookup (fromIntegral i) r of
+          Nothing -> return $ DLVar at m_locvar t i
+          Just s -> return $ DLVar at (Just (at,s)) t i
```

Comment:
This is the same as `return $ DLVar at (M.lookup ....) t i` because you know `m_locvar` is `Nothing`